### ISSUE TITLE
S3 - an invalid signature is generated when the Range header is specified

### ISSUE THREAD

**Issue by nberthet** [akka/alpakka#384]
When the range (https://github.com/akka/alpakka/pull/265) is being added to the download request, it replaces the "Host" header; it causes the generated signature to fail validation on S3.

In S3Stream.scala@101, the original headers should be concatenated
```
      case Some(range) => downloadRequest.withHeaders(downloadRequest.headers :+ headers.Range(range))
```

**Comment by raboof** [comment]
Looks reasonable! You could also use [addHeader](http://doc.akka.io/api/akka-http/current/akka/http/javadsl/model/HttpRequest.html#addHeader(header:akka.http.javadsl.model.HttpHeader):Self) - a PR with this change (and a test) would be most welcome!

**Comment by raboof** [comment]
Oh I missed you already had submitted a PR - unfortunately GitHub linking doesn't always work.

**Comment by johanandren** [comment]
Duplicated and fixed as #460
